### PR TITLE
Issue #13345: Enable FinalParametersCheckExamplesTest

### PR DESCRIPTION
**Pull request by mahfouz72**
issue #13345:
Enables example tests for FinalParametersCheckExamplesTest
I removed the `//ok` comments is it ok. or should I leave them and put an explanation
e.g. `//ok, because it is a primitive type`

### CODE REVIEW

**Comment by mahfouz72** [comment]
@romani does those issues related to my changes? because mvn verify is successful

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example1.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.finalparameters;
+
+// xdoc section -- start
+public class Example1 {
+  public Example1() { }
+  public Example1(final int m) { }
+  public Example1(final int m,int n) { } // violation, 'n should be final'
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example1.java:14]:
```suggestion
  public Example1(final int m, int n) { } // violation, 'n should be final'
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example2.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.finalparameters;
+
+// xdoc section -- start
+public class Example2 {
+  public Example2() { }
+  public Example2(final int m) { }
+  public Example2(final int m,int n) { } // violation, 'n should be final'
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example2.java:16]:
```suggestion
  public Example2(final int m, int n) { } // violation, 'n should be final'
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example3.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.finalparameters;
+
+// xdoc section -- start
+public class Example3 {
+  public Example3() { }
+  public Example3(final int m) { }
+  public Example3(final int m,int n) { }
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example3.java:16]:
```suggestion
  public Example3(final int m ,int n) { }
```

**Review by nrmancuso — CHANGES_REQUESTED** [review]
@mahfouz72 you can disregard the CI that is failing right now, we are aware of the issues.

Few minor items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example3.java (hunk 3 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example3.java]: @nrmancuso done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example1.java (hunk 1 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example1.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example2.java (hunk 2 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example2.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example3.java (hunk 3 above)
Comment (mahfouz72) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example3.java]: I think you mean  `public Example3(final int m , int n) { }` not  `public Example3(final int m  ,int n) { }` and this is done 
please correct me if i am wrong

**Review by nrmancuso — APPROVED** [review]
@mahfouz72 thanks for the quick turnaround! It is good to keep in mind that these examples are our user-facing documentation, so formatting should look good (unless this is the target of the check of course).

**Comment by nrmancuso** [comment]
Rebased on latest master

**Review by romani — APPROVED** [review]

**Pull request merged**